### PR TITLE
Upgrade some deps

### PR DESCRIPTION
**Pull request by dignifiedquire**
This updates the deps that were straightforward to update and some of which `npm audit` was complaining about.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -362,43 +362,43 @@
...
     "minimatch": "^3.0.2",
     "optimist": "^0.6.1",
     "qjobs": "^1.1.4",
     "range-parser": "^1.2.0",
     "rimraf": "^2.6.0",
     "safe-buffer": "^5.0.1",
-    "socket.io": "2.0.4",
-    "source-map": "^0.6.1",
+    "socket.io": "2.1.0",
+    "source-map": "^0.7.2",
```
Comment (thorn0) [inline package.json]: Please note that the `SourceMapConsumer` constructor [became `async`](https://github.com/mozilla/source-map/blob/master/CHANGELOG.md#070) in `source-map` 0.7.0. We should change the code where it's used (`lib/reporter.js`) accordingly.

Comment (johnjbarton) [reply package.json]: Plus the Travis complains:
 `source-map@0.7.2: The engine "node" is incompatible with this module. Expected version ">= 8".`

Comment (dignifiedquire) [reply package.json]: right, there was something.. undid that upgrade

**Comment by tobi-or-not-tobi** [comment]
Is there any planned progress on this PR? The build has failed, perhaps @dignifiedquire  can get some help with this. @johnjbarton can you help or assign someone else?

**Comment by johnjbarton** [comment]
#3044 has just the babel change, and it passes Travis. So I don't know why the last commit here fails.

**Comment by XhmikosR** [comment]
@johnjbarton can you restart the Travis build?

**Comment by lusarz** [comment]
@dignifiedquire 
It would be nice if you update commit message to make travis-ci pass.

**Comment by johnjbarton** [comment]
Unfortunately I used the wrong version of `mime` when I fixed the merge conflicts.

**Comment by johnjbarton** [comment]
close/open to see travis

**Comment by XhmikosR** [comment]
@johnjbarton: just in case you don't know, you should be able to trigger a build via Travis since you have push rights. :)

**Comment by johnjbarton** [comment]
Yes, I've rebuilt this branch many times. It still fails.  I'm continuing to work on improving the Travis tests until this passes.

**Comment by johnjbarton** [comment]
I believe that most or all of the fixes on this PR landed via other PRs after travis issues were resolved. Thanks @dignifiedquire!

**Pull request closed without merging**